### PR TITLE
feat: Bind `modules/<dep_name>/public.ts` to `Module.<depName>`

### PR DESCRIPTION
**Pull request by Blckbrry-Pi**

### CODE REVIEW

**Comment by Blckbrry-Pi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/265?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#265** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/265?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#263** <a href="https://app.graphite.dev/github/pr/rivet-gg/opengb/263?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Blckbrry-Pi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by MasterPtato — APPROVED** [review]

File: src/build/gen/mod.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+function ensureRelativePrefixed(path: string) {
+	if (isAbsolute(path)) return path;
+
+	const parsed = parsePath(path);
+	if (parsed.dir.startsWith(".")) return path;
+	else if (parsed.dir) return formatPath({ ...parsed, dir: `./${parsed.dir}` });
+	else return formatPath({ ...parsed, dir: `.` });
+}
+
+export class GeneratedFile {
```
Comment (NathanFlurry) [inline src/build/gen/mod.ts:25]: nit: move to separate file

nit: rename to something like `GeneratedCodeBuilder` or something with "Builder" in the name

needs docs on what this class does

cool helper though

File: src/build/gen/public_utils.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		// Find the path to the `public.ts` file for this module
+		const modulePublicUtilsPath = resolve(
+			module.path,
+			UTILS_GEN_FILE,
+		);
+
+		// Check if the module has a public utils file
+		const modulePublicUtilsExists = await Deno.stat(modulePublicUtilsPath).then(() => true, () => false);
+
+		if (modulePublicUtilsExists) {
```
Comment (NathanFlurry) [inline src/build/gen/public_utils.ts:32]:
```suggestion
		if (await exists(modulePublicUtilsPath, { isFile: true })) {
```

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
need to include an example in the test repo for this too

File: src/build/gen/public_utils.ts (hunk 2 above)
Comment (Blckbrry-Pi) [reply src/build/gen/public_utils.ts:32]: At least it's kind of poetic that I forgot that `exists` exists?

File: src/build/gen/mod.ts (hunk 1 above)
Comment (Blckbrry-Pi) [reply src/build/gen/mod.ts]: Moved and renamed, will add jsdoc comments

**Review by NathanFlurry — APPROVED** [review]

**Comment by NathanFlurry** [comment]
### Merge activity

* **May 18, 5:00 PM EDT**: @NathanFlurry started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rivet-gg/opengb/265).

**Pull request merged**